thewalrus/tests/test_fock_gradients.py

```diff
@@ -229,3 +231,91 @@ def test_two_mode_squeezing_values(tol):
     T = two_mode_squeezing(r, theta, cutoff)
     expected = ((np.tanh(r) * np.exp(1j * theta)) ** np.arange(cutoff)) / np.cosh(r)
     assert np.allclose(np.diag(T[:, :, 0, 0]), expected, atol=tol, rtol=0)
+
+def test_gaussian_gate_values(tol):
+    """Tests the transforamtion matrix of gaussian gate"""
+    # Single-mode test
+    # Special case: single-mode squeezing (zeta, gamma=0, phi=0)
+    cutoff = 5
+    zeta = 0.3 + 1j * 0.2
+    gamma = 0
+    phi = 0
+    r = np.abs(zeta)
+    delta = np.angle(zeta)
+    expected = squeezing(r, delta, cutoff)
+    tanhr = np.tanh(r)
+    sechr = 1 / np.cosh(r)
+    C = np.sqrt(sechr)
+    mu = np.zeros(2).T
+    Sigma = np.array([[np.exp(1j * delta) * tanhr, -sechr], [-sechr, -np.exp(-1j * delta) * tanhr]])
+    T = gaussian_gate(C, mu, Sigma, cutoff, 1)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
+    # Special case: single-mode displacement (gamma, zeta=0, phi=0)
+    cutoff = 4
+    gamma = 0.2 - 1j * 0.8
+    zeta = 0
+    phi = 0
+    expected = displacement(np.abs(gamma), np.angle(gamma), cutoff)
+    C = np.exp(-0.5 * np.abs(gamma) ** 2)
+    mu = np.array([gamma, -np.conj(gamma)]).T
+    Sigma = np.array([[0, -1], [-1, 0]])
+    T = gaussian_gate(C, mu, Sigma, cutoff, 1)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
+    # Special case: BS
+    cutoff = 4
+    theta = np.pi / 4
+    phi = np.pi / 2
+    expected = beamsplitter(theta, phi, cutoff)
+    ct = np.cos(theta)
+    st = np.sin(theta) * np.exp(1j * phi)
+    V = np.array([[ct, -np.conj(st)], [st, ct]])
+    C = 1
+    mu = np.zeros(4).T
+    Sigma = -np.block([[np.zeros((2, 2)), V], [V.T, np.zeros((2, 2))]])
+    T = gaussian_gate(C, mu, Sigma, cutoff, 2)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
+
+
+def test_grad_gaussian_gate(tol):
+    """Tests the gradients of the transforamtion matrix of gaussian gate"""
```

Comment:
I would suggest you break this tests into two, since you are testing two different gates.
It would also be nice to add test comparing `gaussian_gate` and `fock_tensor`.

Reply:
I'm writing the test code to compare with `fock_tensor` in SF. Because as I understand, `fock_tensor` gives the final gaussian state after the transformation, but here we only have the tensor/matrix for the transformation.

Reply:
I'm also testing three gates for `test_gaussian_gate_values` to make sure the special cases... Do you think I need also to split that function?

Reply:
Hi @sylviemonet ---- `fock_tensor` gives the tensor/matrix representation of the transformation.
As for your second question it would be simpler if you split the test into three.